src/apps/content-editor/src/app/ContentEditor.js

```diff
@@ -21,75 +25,100 @@ import { CSVImport } from "./views/CSVImport";
 import "@zesty-io/core/vendor.css";
 
 import styles from "./ContentEditor.less";
-export default connect((state) => {
-  return {
-    contentModels: state.models,
-    navContent: state.navContent,
+export default function ContentEditor(props) {
+  const contentModels = useSelector((state) => state.models);
+  const navContent = useSelector((state) => state.navContent);
+  const ui = useSelector((state) => state.ui);
+  const dispatch = useDispatch();
+  const [mouseEnterTimer, setMouseEnterTimer] = useState(null);
+  const [mouseLeaveTimer, setMouseLeaveTimer] = useState(null);
+
+  const handleMouseEnter = () => {
+    const enterTimer = setTimeout(() => {
+      dispatch(actions.setContentNavHover(true));
+    }, 500);
+
+    setMouseEnterTimer(enterTimer);
+  };
+
+  const handleMouseLeave = () => {
+    const leaveTimer = setTimeout(() => {
+      dispatch(actions.setContentNavHover(false));
+    }, 500);
+    setMouseLeaveTimer(leaveTimer);
+
+    clearTimeout(mouseEnterTimer);
+    clearTimeout(mouseLeaveTimer);
   };
-})(
-  class ContentEditor extends Component {
-    componentDidMount() {
-      // Kick off loading data before app mount
-      // to decrease time to first interaction
-      this.props.dispatch(fetchNav());
-      this.props.dispatch(fetchModels());
-    }
-    render() {
-      return (
-        <WithLoader
-          condition={
-            this.props.navContent.nav.length ||
-            this.props.navContent.headless.length
-          }
-          message="Starting Content Editor"
+
+  useEffect(() => {
+    // Kick off loading data before app mount
+    // to decrease time to first interaction
+    dispatch(fetchNav());
+    dispatch(fetchModels());
+  }, []);
+
+  return (
+    <WithLoader
+      condition={navContent.nav.length || navContent.headless.length}
+      message="Starting Content Editor"
+    >
+      <section
+        className={cx(
+          styles.ContentEditor,
+          ui.contentNav ? styles.ContentNavOpen : "",
+          ui.contentNavHover && !ui.contentNav ? styles.ContentNavHover : ""
+          // true ? styles.ContentNavHover : ""
+        )}
+      >
+        <div
```

Comment:
Move into the `ContentNav` component. The `ContentEditor` should remain strictly layout.